internal/flows/upgrade.go

```diff
@@ -2,22 +2,131 @@ package flows
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/aws/eks-hybrid/internal/aws"
+	"github.com/aws/eks-hybrid/internal/cni"
+	"github.com/aws/eks-hybrid/internal/containerd"
+	"github.com/aws/eks-hybrid/internal/creds"
+	"github.com/aws/eks-hybrid/internal/daemon"
+	"github.com/aws/eks-hybrid/internal/iamauthenticator"
+	"github.com/aws/eks-hybrid/internal/iamrolesanywhere"
+	"github.com/aws/eks-hybrid/internal/imagecredentialprovider"
+	"github.com/aws/eks-hybrid/internal/iptables"
+	"github.com/aws/eks-hybrid/internal/kubectl"
+	"github.com/aws/eks-hybrid/internal/kubelet"
+	"github.com/aws/eks-hybrid/internal/nodeprovider"
+	"github.com/aws/eks-hybrid/internal/packagemanager"
+	"github.com/aws/eks-hybrid/internal/ssm"
+	"github.com/aws/eks-hybrid/internal/tracker"
 )
 
 type Upgrader struct {
-	*Uninstaller
-	*Installer
-	*Initer
+	NodeProvider       nodeprovider.NodeProvider
+	AwsSource          aws.Source
+	PackageManager     *packagemanager.DistroPackageManager
+	CredentialProvider creds.CredentialProvider
+	Artifacts          *tracker.InstalledArtifacts
+	DaemonManager      daemon.DaemonManager
+	SkipPhases         []string
+	Logger             *zap.Logger
 }
 
 func (u *Upgrader) Run(ctx context.Context) error {
-	if err := u.Uninstaller.Run(ctx); err != nil {
+	if err := u.upgradeDistroPackages(ctx); err != nil {
+		return err
+	}
+
+	if err := u.upgradeCredentialProvider(ctx); err != nil {
+		return err
+	}
+
+	if err := u.upgradeEksArtifacts(ctx); err != nil {
+		return err
+	}
+
```

Comment:
do we need to call `ValidateConfig` here? or even before

Reply:
We run it much before since we need to fast fail in the cmd package itself. I'm re-thinking about the validations we have in the cmd package too, I think that needs to be organized well. But thats for another PR.